openstack/loadbalancer/v2/providers/requests.go

```diff
@@ -0,0 +1,18 @@
+package providers
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// List returns a Pager which allows you to iterate over a collection of
+// providers.
+//
+// Default policy settings return only those providers that are owned by
+// the project who submits the request, unless an admin user submits the request.
+func List(c *gophercloud.ServiceClient) pagination.Pager {
```

Comment:
It looks like the List call can accept an option to limit the returned fields:

```go
type ListOptsBuilder interface {
	ToProviderListQuery() (string, error)
}

type ListOpts struct {
  Fields []string `q:"fields"`
}

func (opts ListOpts) ToProviderListQuery() (string, error) {
  	q, err := gophercloud.BuildQueryString(opts)
	if err != nil {
		return "", err
	}
	return q.String(), nil
}

func List(c *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
```

Reply:
right! I'll add this!